### PR TITLE
refactor(project landing page): add interfaces for metadata object (DSP-1183)

### PR DESCRIPTION
**Pull request by waychal**
resolves DSP-1183

In this PR, we use interfaces to define metadata object.

### CODE REVIEW

File: src/app/project/board/dataset-metadata.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
+/* Interface to for url type */
```
Comment (flavens) [inline src/app/project/board/dataset-metadata.ts:1]: typo

File: src/app/project/board/attribution-tab-view/attribution-tab-view.component.html [hunk 2]
```diff
@@ -4,11 +4,17 @@ <h1>Attribution</h1>
...
 <div class="tab-contents">
 
-    <mat-card *ngFor="let entry of attribution | keyvalue;" class="attribution-entry">
+    <mat-card *ngFor="let attribution of attributions" class="attribution-entry">
 
-        <h3 class="color-for-role"> {{ entry.value['role'] | titlecase }} </h3>
+        <h3 class="color-for-role"> {{ attribution.role | titlecase }} </h3>
 
-        <app-person-template [person]="people[entry.key]"></app-person-template>
+        <div *ngIf="attribution.agent.familyName">
```
Comment (flavens) [inline src/app/project/board/attribution-tab-view/attribution-tab-view.component.html:11]: in my code editor, I get the following error message: "Identifier 'familyName' is not defined. 'IPerson | IOrganisation' does not contain such a member". It is the same for `attribution.agent.url`

Comment (waychal) [reply src/app/project/board/attribution-tab-view/attribution-tab-view.component.html:11]: I see this error in VC but not in webstorm editor. 

`attribution.agent`is of type `IPerson | IOrganisation` in interface. `familyName` is a property of `IPerson` interface and `url` is optional property of `IOrganisation`. I am trying to understand why it is still giving error in VC.

File: src/app/project/board/dataset-metadata.ts (hunk 1 above)
Comment (waychal) [reply src/app/project/board/dataset-metadata.ts:1]: Done.

File: src/app/project/board/attribution-tab-view/attribution-tab-view.component.html (hunk 2 above)
Comment (waychal) [reply src/app/project/board/attribution-tab-view/attribution-tab-view.component.html]: Done.

**Review by flavens — APPROVED** [review]

File: src/app/project/board/dataset-metadata.ts [hunk 3]
```diff
@@ -0,0 +1,123 @@
+/* Interface for url type */
```
Comment (mpro7) [inline src/app/project/board/dataset-metadata.ts:1]: I would remove all comments for interfaces which are self-descriptive and leave only these which are not.

**Review by mpro7 — COMMENTED** [review]
I'm too late but @waychal you can reflect my comments in next task. Thanks!

File: src/app/project/board/attribution-tab-view/attribution-tab-view.component.ts [hunk 4]
```diff
@@ -1,82 +1,22 @@
...
+    getAgentType (agent: IPerson | IOrganisation) {
+        if (agent.hasOwnProperty('familyName')){
+            return 'person';
         }
-    }
-
+        if (agent.hasOwnProperty('url')){
+            return 'organisation';
+        } 
+    };
```
[the remark this answers is not in the document]
Comment (waychal) [reply src/app/project/board/attribution-tab-view/attribution-tab-view.component.ts:21]: Done

File: src/app/project/board/project-tab-view/project-tab-view.component.html [hunk 5]
```diff
@@ -13,68 +13,55 @@ <h3 class="label mat-subheading-1">
 
                     <!-- #################### template for discipline ################# -->
```
[the remark this answers is not in the document]
Comment (waychal) [reply src/app/project/board/project-tab-view/project-tab-view.component.html:14]: Done

File: src/app/project/board/dataset-metadata.ts (hunk 3 above)
Comment (waychal) [reply src/app/project/board/dataset-metadata.ts:1]: Done

**Comment by waychal** [comment]
@mpro7 I have made the changes in in another [PR](https://github.com/dasch-swiss/dsp-app/pull/368#pullrequestreview-573357078)

**Pull request merged**